packages/uicore/src/components/ExpressionInput/utils.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { ChildKeyNode } from 'components/ExpressionDropdown/ExpressionDropdown'
+
+interface TrieNode {
+  value: string
+  valueTillHere: string
+  children: TrieNode[]
+  childKeys: ChildKeyNode[]
+  childExpressions: string[]
+}
+
+export function getDataStructureInCorrectFormat(expressions: string[]): TrieNode {
+  const rootTrieNode: TrieNode = {
+    value: 'Main',
+    valueTillHere: '',
+    children: [],
+    childKeys: [],
+    childExpressions: []
+  }
+  expressions.forEach(expression => {
+    const expressionWords = expression.split('.')
+    let startNode = rootTrieNode
+
+    expressionWords.forEach((word, index) => {
+      startNode.childExpressions.push(expression)
+      const isWordPresent = startNode.children.findIndex(child => child.value === word)
```

Comment:
using 'value' as key would have been better here to avoid findIndex 😛

Reply:
Will think about this :p